src/constraint_aware_append.c

```diff
@@ -160,11 +160,26 @@ ca_append_begin(CustomScanState *node, EState *estate, int eflags)
 
 	forboth(lc_plan, old_appendplans, lc_info, append_rel_info)
 	{
-		Scan	   *scan = lfirst(lc_plan);
-		AppendRelInfo *appinfo = lfirst(lc_info);
-		RangeTblEntry *rte = rt_fetch(scan->scanrelid, estate->es_range_table);
+		Node	   *node = (Node *) lfirst(lc_plan);
```

Comment:
Better to cast this to `Plan`, since all nodes here should be a plan. Then you don't need all the casting below.